pints/tests/test_mcmc_controller.py

```diff
@@ -544,6 +544,29 @@ def test_deprecated_alias(self):
             self.log_posterior, 1, [self.real_parameters])
         self.assertIsInstance(mcmc, pints.MCMCController)
 
+    def test_post_run_statistics(self):
+        # Test method to obtain post-run statistics
+
+        # Set up test problem
+        x0 = np.array(self.real_parameters) * 1.05
+        x1 = np.array(self.real_parameters) * 1.15
+        x2 = np.array(self.real_parameters) * 0.95
+        xs = [x0, x1, x2]
+
+        mcmc = pints.MCMCController(self.log_posterior, len(xs), xs)
+        mcmc.set_initial_phase_iterations(5)
+        mcmc.set_max_iterations(10)
+        mcmc.set_log_to_screen(False)
+        mcmc.set_log_to_file(False)
+
+        # Before run, methods return None
+        self.assertIsNone(mcmc.time())
+
+        # Check post-run output
+        mcmc.run()
+        self.assertIsInstance(mcmc.time(), float)
+        self.assertGreater(mcmc.time(), 0)
```

Comment:
Hmmm, unless we use exactly the same timing method (and there are many, unfortunately), that could give quite different results? Esp. when running on some virtual machine I think, imagine if processing gets halted for whatever reason - some other process takes priority and uses up all cpu power for a few seconds...
I've had to remove tests like `time_simple_thing < time_hard_thing` in the past because they failed occasionally in CI !

Reply:
Now checking upper and lower bounds